src/modules/extrakeys/tests_impl.h

```diff
@@ -643,84 +645,82 @@ void permute(size_t *arr, size_t n) {
     }
 }
 
-void rand_xonly_pk(secp256k1_xonly_pubkey *pk) {
+void rand_pk(secp256k1_pubkey *pk) {
     unsigned char seckey[32];
     secp256k1_keypair keypair;
     secp256k1_testrand256(seckey);
     CHECK(secp256k1_keypair_create(ctx, &keypair, seckey) == 1);
-    CHECK(secp256k1_keypair_xonly_pub(ctx, pk, NULL, &keypair) == 1);
+    CHECK(secp256k1_keypair_pub(ctx, pk, &keypair) == 1);
 }
 
-void test_xonly_sort_api(void) {
+void test_sort_api(void) {
     int ecount = 0;
-    secp256k1_xonly_pubkey pks[2];
-    const secp256k1_xonly_pubkey *pks_ptr[2];
+    secp256k1_pubkey pks[2];
+    const secp256k1_pubkey *pks_ptr[2];
     secp256k1_context *none = api_test_context(SECP256K1_CONTEXT_NONE, &ecount);
 
     pks_ptr[0] = &pks[0];
     pks_ptr[1] = &pks[1];
 
-    rand_xonly_pk(&pks[0]);
-    rand_xonly_pk(&pks[1]);
+    rand_pk(&pks[0]);
+    rand_pk(&pks[1]);
 
-    CHECK(secp256k1_xonly_sort(none, pks_ptr, 2) == 1);
-    CHECK(secp256k1_xonly_sort(none, NULL, 2) == 0);
+    CHECK(secp256k1_pubkey_sort(none, pks_ptr, 2) == 1);
+    CHECK(secp256k1_pubkey_sort(none, NULL, 2) == 0);
     CHECK(ecount == 1);
-    CHECK(secp256k1_xonly_sort(none, pks_ptr, 0) == 1);
+    CHECK(secp256k1_pubkey_sort(none, pks_ptr, 0) == 1);
     /* Test illegal public keys */
     memset(&pks[0], 0, sizeof(pks[0]));
-    CHECK(secp256k1_xonly_sort(none, pks_ptr, 2) == 1);
+    CHECK(secp256k1_pubkey_sort(none, pks_ptr, 2) == 1);
     CHECK(ecount == 2);
     memset(&pks[1], 0, sizeof(pks[1]));
-    CHECK(secp256k1_xonly_sort(none, pks_ptr, 2) == 1);
+    CHECK(secp256k1_pubkey_sort(none, pks_ptr, 2) == 1);
     CHECK(ecount > 2);
 
     secp256k1_context_destroy(none);
 }
 
-void test_xonly_sort(void) {
-    secp256k1_xonly_pubkey pk[5];
-    unsigned char pk_ser[5][32];
+void test_sort(void) {
+    secp256k1_pubkey pk[5];
+    unsigned char pk_ser[5][33] = {
+        { 0x02, 0x08 },
+        { 0x02, 0x0b },
+        { 0x02, 0x0c },
+        { 0x03, 0x05 },
+        { 0x03, 0x0a },
```

Comment:
I know this is not this PR but I think It would really be nice to have some more variety here in the test cases here. For example, if you had forgotten to replace 32 by 33 in the comparison function in this PR, the tests wouldn't have caught this so maybe add at least a test case where the keys differ only the last byte. 

edit: ok now that I see the next commit, maybe we should add such a case to the test vectors... This will be better but this is certainly not in the scope of this PR. (So feel free to address this in test_sort or ignore it here and leave it leave entirely to an improved test vector.)

Reply:
Good point. I added a test vector to https://github.com/jonasnick/bips/pull/75 and also added it here.